src/Http/Cookie/Cookie.php

```diff
@@ -236,10 +236,22 @@ public function getValue()
     }
 
     /**
-     * {@inheritDoc}
-     * @psalm-suppress InvalidReturnType
+     * Gets the cookie value as a string.
+     *
+     * This will collapse any complex data in the cookie with json_encode()
+     *
+     * @return mixed
+     * @deprecated 4.0.0 Use getScalarValue() instead.
      */
     public function getStringValue()
+    {
+        return $this->getScalarValue();
```

Comment:
`return (string)$this->getScalarValue();`?

aka could this not be an integer? The method name suggests otherwise.

Reply:
The old method name was misleading, hence it's deprecated and new method added instead.